docs/backup_and_restore.rst

```diff
@@ -123,67 +105,213 @@ Restoring
-``install_files/ansible-base`` under the SecureDrop repository root directory.
-If you are using the same *Admin Workstation* to do a restore from a previous
-backup, it should already be there because it was placed there by the backup
-command. Otherwise, you should copy the backup archive that you wish to restore to
-``install_files/ansible-base``.
+If you are restoring data onto an existing instance (for example, for data 
+recovery purposes), see 
+:ref:`Restoring Data on an Existing SecureDrop Installation <restore_data>`.
 
-.. note:: The backup strategy used for SecureDrop versions prior to 0.3.7
-          created encrypted archives with the extension ``.zip.gpg``.
-          You can safely remove those files once you've created the ``.tar.gz``
-          backup archive described in this guide.
+If you are reinstalling SecureDrop and then restoring from a backup (for 
+example, for hardware migration or disaster recovery purposes), see 
+:ref:`Migrating <migrating>`.
 
-Restoring From a Backup File
-''''''''''''''''''''''''''''
+.. _restore_data:
 
-.. important:: This documentation applies to a SecureDrop instance using
-               v2 onion services. If your instance uses v3 onion services,
-               you will need to follow additional steps depending on your
-               specific restore scenario.
+Restoring Data on an Existing SecureDrop Installation
+''''''''''''''''''''''''''''''''''''''''''''''''''''' 
 
-To perform a restore, you must already have a backup archive. Provide its
-filename in the following command:
+To restore an existing instance to a previous state, run the command:
 
 .. code:: sh
 
-   ./securedrop-admin restore sd-backup-2017-07-22--01-06-25.tar.gz
+   ./securedrop-admin restore sd-backup-2020-07-22--01-06-25.tar.gz
 
-Make sure to replace ``sd-backup-2017-07-22--01-06-25.tar.gz`` with the filename
+Make sure to replace ``sd-backup-2020-07-22--01-06-25.tar.gz`` with the filename
 for your backup archive. The backup archives are located in
 ``install_files/ansible-base``.
 
-Once the restore is done, the *Application Server* will use the original
-*Source Interface* and *Journalist Interface* Onion URLs. You will need
-to update the corresponding files on the *Admin Workstation*:
+This command restores submissions, source and journalist accounts, and Tor
+configuration (*Source* and *Journalist Interface* onion URLs) to their state at the 
+time of the backup.
+
+.. _migrating:
+
+Migrating
+---------
+
+.. note:: These steps require physical access to the servers. 
+         
+Moving a SecureDrop installation to new hardware consists of:
+
+  - Backing up the existing installation and preserving its Tor credentials from the *Admin Workstation* 
+  - Installing the same version of SecureDrop on the new hardware;
+  - Restoring the backup to the new installation;
+  - Repairing Tor credentials for the new installation. 
+
+Detailed Migration Instructions
```

Comment:
These instructions are a bit complex and branching - it might make sense to split them into 3 sections as follows:

1) v3-only - this should be simple: 
- copy `tor_v3_keys.json` to the fresh  clone of securedrop, 
- then proceed with a fresh install 
- and then restore from the backup

2) v2+v3 - actually, this will be the same as v3-only, as who cares about the v2 ssh setup, it's not used. You can copy app's `*ths` files as below if you want the `ssh app-legacy` and `ssh mon-legacy` commands to work
3) v2-only - this one is probably the most complex - you will need to 
- copy app's `*ths` files from old install, do a fresh install, 
- restore from backup (which will fail with an error that says to run tailsconfig)
- copy app's `*ths` files into place, 
- run tailsconfig, 
- then run restore again. 

The trick here is that you don't have to rebuild mon's config for v2, you can just set up a new ssh config. Also for the configs with a v3 element, all you should need is the `tor_v3_keys.json` file.

(Also I may be missing steps here, it's been a while!)

Reply:
Ooh also you'll need to copy keys from `securedrop.old` and you might as well copy `site-specific` while you're in there.

Reply:
Or you could skip the new clone and just move files you don't want out of the way....

Reply:
So, what's the scenario in which we want to support them preserving v2 as part of a migration/reinstall? Should we just make it explicit that they should migrate to v3 while they're there? These docs didn't use to exist at all, if we add them now, it feels like we might as well make them v3-only.

Reply:
Hypothetically? Someone may have a downed v2 instance with an old backup available, and they may want to restore it before migrating. Or they have some other reason unique to them for going right up to the wire on v2. But adding a "Data-only migration to v3" option might not be a bad idea. something like:
---
If you want to restore the data from a backup on a fresh instance with new onion service addresses:

- complete the fresh install first
- copy the backup file FILENAME to install_files/ansible base
- run `./securedrop-admin restore --preserve-tor-config FILENAME`
---
would do the trick. (this is v2/v3 only but if we disable new v2 installs next release that's a strong hint.)

Reply:
So the "data-only migration to v3" section is in here, more opaquely referred to as  "Restoring Data Without Restoring Tor Configuration", and is for the case of instances with a v2 backup file who want to recover data (ahem, if certain deadlines are missed). 

I think there's an argument for skipping the v2-only hardware migration instructions, and telling users to contact Support. As soon as we stop allowing v2 onion services on fresh installs, the v2 migration instructions won't work. However, since they are written up I'm including them for now anyone says otherwise, and we can remove them in a few weeks' time.

Reply:
@rocodes: 
> However, since they are written up I'm including them for now 

Sounds reasonable; perhaps we can just add a blurb at the top of the section, e.g.:

> If you are migrating or restoring an installation using v2 onion services, please be aware that support for them will be removed from SecureDrop starting in March 2021. We encourage you to migrate to v3 onion services as part of any migration. To do so, reinstall SecureDrop with v3 onion services enabled and restore your data without overwriting the generated Tor configuration **[link to section]**. The instructions for users of v2 onion services below will be removed in a future release.

Reply:
Added in note on depreciation 254b0d9, and also split out migration instructions by onion services version.